app/src/main/java/com/example/bookclub/QRCodeImageAnalyzer.java

```diff
@@ -0,0 +1,56 @@
+package com.example.bookclub;
+
+import static android.graphics.ImageFormat.YUV_420_888;
+import static android.graphics.ImageFormat.YUV_422_888;
+import static android.graphics.ImageFormat.YUV_444_888;
+
+import androidx.annotation.NonNull;
+import androidx.camera.core.ImageAnalysis;
+import androidx.camera.core.ImageProxy;
+
+import com.example.bookclub.utils.QRCodeFoundListener;
+import com.google.zxing.BinaryBitmap;
+import com.google.zxing.ChecksumException;
+import com.google.zxing.FormatException;
+import com.google.zxing.NotFoundException;
+import com.google.zxing.PlanarYUVLuminanceSource;
+import com.google.zxing.Result;
+import com.google.zxing.common.HybridBinarizer;
+import com.google.zxing.multi.qrcode.QRCodeMultiReader;
```

Comment:
Are we scanning a QR Code or a Barcode?

Reply:
We scanning for QR code. The branch name is confusing I am sorry for that.